pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -538,11 +558,15 @@ func Run() {
 			agentlog.RebootReason(errStr)
 			execReboot(true)
 			agentlog.CheckMaxTime(agentName, start)
+
+		case <-stillRunning.C:
+			doDeviceReboot(&zedagentCtx)
```

Comment:
This is odd. stillRunning is about the touch file. Do you need that? If not pick a different timer name. But see other comment about deviceReboot boolean.

Reply:
sure